static/js/recent_topics.js

```diff
@@ -0,0 +1,79 @@
+var recent_topics = (function () {
+
+var exports = {};
+
+var topics = new Dict(); // Key is stream-id:subject.
+
+exports.process_all_messages = function () {
+    exports.process_messages(message_list.all.all_messages());
+};
+
+exports.process_messages = function (messages) {
+    topics = new Dict();
+    messages.forEach(exports.process_message);
+};
+
+exports.process_message = function (msg) {
+    var is_ours = msg.sender_id === page_params.user_id;
+    var is_relevant = is_ours && msg.type === 'stream';
+    var key = msg.stream_id + ':' + msg.subject;
+    var topic = topics.has(key) ? topics.get(key) : undefined;
```

Comment:
You can simplify this to `var topic = topics.get(key)`, which will return `undefined` if the key doesn't exist.